aten/src/ATen/ParallelNative.h

```diff
@@ -60,7 +60,10 @@ inline void parallel_for(
       internal::_unset_thread_num();
     };
 
-    std::vector<c10::ivalue::Future> futures(num_tasks);
+    std::vector<std::shared_ptr<c10::ivalue::Future>> futures(num_tasks);
```

Comment:
shared_ptr here is not necessary. It can be done with emplace_back.

Reply:
I'd prefer a non-shared_ptr version too

Reply:
not quite out of the box -- the implicit move constructor is ill-formed (std::mutex etc).  I don't think it's worth writing + maintaining.